internal/services/bot/bot_channel_ms_teams_resource.go

```diff
@@ -63,13 +71,34 @@ func resourceBotChannelMsTeams() *pluginsdk.Resource {
 				ValidateFunc: validate.BotMSTeamsCallingWebHook(),
 			},
 
+			"deployment_environment": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Default:  "CommercialDeployment",
+				ValidateFunc: validation.StringInSlice([]string{
+					"CommercialDeployment",
+					"GCCModerateDeployment",
+				}, false),
+			},
+
 			// TODO 4.0: change this from enable_* to *_enabled
 			"enable_calling": {
 				Type:     pluginsdk.TypeBool,
 				Optional: true,
 				Default:  false,
 			},
 		},
+
+		CustomizeDiff: pluginsdk.CustomDiffWithAll(
+			func(ctx context.Context, d *schema.ResourceDiff, meta interface{}) error {
+				oldVal, newVal := d.GetChange("accepted_terms_enabled")
+				if oldVal.(bool) && !newVal.(bool) {
+					return fmt.Errorf("`accepted_terms_enabled` cannot be disabled once it's been enabled")
```

Comment:
re-creating this resource would presumably enable this, so why are we using a `customizeDiff` here?

Reply:
@tombuildsstuff , This property cannot be changed once it's enabled. Even if azurerm_bot_channel_ms_teams is destroyed, but this property still cannot be updated back to "false" once it's enabled. Unless the parent resource azurerm_bot_channels_registration is destroyed, then this property can be set back to "false".

Reply:
Service team confirmed that they expect users always set accepted_terms_enabled to true. So I hardcoded to true for this property.

Reply:
Service team confirmed that they expect users always set it to true. So I hardcoded it to true.